build.cake

```diff
@@ -0,0 +1,183 @@
+            var sample = System.IO.Path.GetFileNameWithoutExtension(proj);
+            DisplayHeading("Building " + sample + " sample");
+            BuildProject(proj, configuration);
+        }
+        
+    });
+
+//////////////////////////////////////////////////////////////////////
+// TEST
+//////////////////////////////////////////////////////////////////////
+
+Task("Test")
+.IsDependentOn("Rebuild")
+.OnError(exception => ErrorDetail.Add(exception.Message))
+.Does(() =>
+    {
+        foreach(var proj in ProjList)
+        {
+            var bin = DirFrom(proj) + "/bin/";
+            var projName = System.IO.Path.GetFileNameWithoutExtension(proj);
+            var dllName = bin + projName + ".dll";
+
+            DisplayHeading("Testing " + projName + " sample");
+
+            try
+            {
+                NUnit3(dllName);
+            }
+            catch(CakeException ex)
+            {
+                ErrorDetail.Add("     * " + projName + " test failed.");
+            }
+
+        }
+    });
+
+//////////////////////////////////////////////////////////////////////
+// TEARDOWN TASK
+//////////////////////////////////////////////////////////////////////
+
+Teardown(() =>
+    {
+        CheckForError(ref ErrorDetail);
+    });
+
+void CheckForError(ref List<string> errorDetail)
+{
+    if(errorDetail.Count != 0)
+    {
+        var copyError = new List<string>();
+        copyError = errorDetail.Select(s => s).ToList();
+        errorDetail.Clear();
+        throw new Exception("One or more unit tests failed, breaking the build.\n"
+            + copyError.Aggregate((x,y) => x + "\n" + y));
+    }
+}
+
+//////////////////////////////////////////////////////////////////////
+// HELPER METHODS
+//////////////////////////////////////////////////////////////////////
+
+void BuildProject(string projPath, string configuration)
+{
+    if (IsRunningOnWindows())
+    {
+        MSBuild(projPath, new MSBuildSettings()
+            .SetConfiguration(configuration)
+            .SetMSBuildPlatform(MSBuildPlatform.Automatic)
+            .SetVerbosity(Verbosity.Minimal)
+            .SetNodeReuse(false));
+    }
+    else
+    {
+        XBuild(projPath, new XBuildSettings()
+            .WithTarget("Build")
+            .WithProperty("Configuration", configuration)
+            .SetVerbosity(Verbosity.Minimal));
+    }
+}
+
+string DirFrom(string filePath)
+{
+    return System.IO.Path.GetDirectoryName(filePath);
+}
+
+void DisplayHeading(string heading)
+{
+    Information("");
+    Information("----------------------------------------");
+    Information(heading);
+    Information("----------------------------------------");
+    Information("");
+}
+
+
+//////////////////////////////////////////////////////////////////////
+// TASK TARGETS
+//////////////////////////////////////////////////////////////////////
+
+Task("Rebuild")
+.IsDependentOn("Clean")
+.IsDependentOn("Build");
+
+Task("Default")
+.IsDependentOn("Rebuild");
```

Comment:
Had left "Appveyor" out as it seemed redundant for samples, but will put it in for consistency =)

> Our other scripts use Build as default

I saw, although nunit/build.cake has the comment:
```` 
Task("Default")
    .IsDependentOn("Build"); // Rebuild?
````
which was why I used Rebuild here. Seems preferable to me that if you run a build script which fails, you can't accidently then work on an old build. I should probably have highlighted that for discussion, slipped my mind, sorry.



> Also, "build" is the name of the command.

Not sure what you mean there, sorry?